referenceframe/model.go

```diff
@@ -169,6 +169,7 @@ func (m *SimpleModel) CachedTransform(inputs []Input) (spatialmath.Pose, error)
 func (m *SimpleModel) DoF() []Limit {
 	m.lock.RLock()
 	if len(m.limits) > 0 {
+		defer m.lock.RUnlock()
 		return m.limits
 	}
 	m.lock.RUnlock()
```

Comment:
why not defer it right before we lock?

Reply:
Then we would be double unlocking it if the len is 0

Reply:
nvm ray explain to me